src/Scripts/VirtualScript.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { VirtualScript, VirtualScriptType } from './VirtualScript';
+import { expect, use } from 'chai';
+import promisified from 'chai-as-promised';
+import Module from 'module';
+
+use(promisified);
+
+describe('VirtualScript', () => {
+  const identityScript = `
+  const handle = (...args) => {
+    return args;
+  };
+  exports.handle = handle;
+  `;
+
+  const createAndCompileVirtualScript = (code: string) => {
+    const virtualScript = new VirtualScript(
+      '123',
+      VirtualScriptType.LOCAL,
+      code
+    );
+    virtualScript.compile();
+
+    return virtualScript;
+  };
+
+  it('should throw when constructed with falsy id', () => {
+    expect(
+      () => new VirtualScript('', VirtualScriptType.REMOTE, 'let a = 1;')
+    ).to.throw();
+    expect(
+      () => new VirtualScript(null, VirtualScriptType.REMOTE, 'let a = 1;')
+    ).to.throw();
+    expect(
+      () => new VirtualScript(undefined, VirtualScriptType.REMOTE, 'let a = 1;')
+    ).to.throw();
+  });
```

Comment:
You should consider them as separate test cases
```suggestion
  ['', undefined, null].forEach((input) => it(`should throw an error if "${input}" is supplied as ID`, () => {
    expect(
      () => new VirtualScript(input, VirtualScriptType.REMOTE, 'let a = 1;')
    ).to.throw();
  }));
```